pkg/core/blockchain.go

```diff
@@ -187,38 +193,32 @@ func (bc *Blockchain) Run(ctx context.Context) {
 	}
 }
 
-// AddBlock processes the given block and will add it to the cache so it
-// can be persisted.
+// AddBlock accepts successive block for the Blockchain, verifies it and
+// stores internally. Eventually it will be persisted to the backing storage.
 func (bc *Blockchain) AddBlock(block *Block) error {
-	if !bc.blockCache.Has(block.Hash()) {
-		bc.blockCache.Add(block.Hash(), block)
+	expectedHeight := bc.BlockHeight() + 1
```

Comment:
just curios: why don't we have check for the BlochHeight inside the bc.verfyBlocks? 
It looks like verification logic but we put it separately here.

Reply:
after the comment realized:
bc.verifyBlocks === bc.ShouldVeriffyBlocks (it just a flag)

and block.Verify check block itself.
Now it's clear for me. nvrmd first comment